third_party/nvidia/backend/compiler.py

```diff
@@ -229,6 +229,11 @@ def make_llir(src, metadata, options, capability):
         # TritonGPU -> LLVM-IR (MLIR)
         pm = ir.pass_manager(mod.context)
         pm.enable_debug()
+        # Set up Diagnostic
+        if os.environ.get("MLIR_ENABLE_REMARK", "0") == "1":
+            srcMgr = llvm.source_mgr()
+            diag = ir.source_mgr_diag(srcMgr, mod.context)
+            mod.context.printOpOnDiagnostic(True)
```

Comment:
Is it possible to emit all diagnostic messages to a file? At least have such an option even if the default is stdout or stderr.

Reply:
@manman-ren We may also want to design a warning format that could be parsed by proton

Reply:
@Jokeren Yes, we need a warning format. Do you have any suggestion? Do we want to mention the pass name? Or add a severity level?

Reply:
> Is it possible to emit all diagnostic messages to a file? At least have such an option even if the default is stdout or stderr.

I think it is possible. Do we want an env variable to supply the file path?

Reply:
I don't have a format in mind now, but I think you could use a function to concatenate the warning with a dedicated separator.

For instance, emitWarning(ir, passName/fileName, map<operands/results, problem>)

Reply:
Yeah we need to talk more about writing to a file and the format of the diagnostics. Let's do that outside of this PR. Zeng and I looked at writing to a file offline: SourceMgrDiagnosticHandler can take a raw_ostream (the default is stderr).